text/0000-decorators.md

```diff
@@ -18,9 +18,85 @@ This RFC is being made now so that we can get early discussion out of the way, a
 
 Alternatively, if decorators do not move forward in TC39 in the next few months, we may need to consider making an exception for decorators, or scoping down Ember Octane.
 
+## Terminology
+
+For the purposes of this RFC, we'll use the following terminology:
+
+* The **Octane programming model** refers to the new programming model established by the Ember Octane edition. It includes _native classes_, _tracked properties_ and _glimmer components_, and more generally refers to features that will be considered _core to Ember_ in the future.
+* The **classic programming model** refers to the traditional programming model. It includes _classic classes_, _computed properties_, _event listeners_, _observers_, _property notifications_, and _curly components_, and more generally refers to features that will _eventually be removed from Ember_.
+* **Native classes** are classes defined using the Javascript `class` keyword
+* **Classic classes** are classes defined by subclassing from `EmberObject` using the static `extend` method.
+* **Tracked properties** refer to a new method of change tracking which is part of the Octane programming model. They will be proposed in a separate RFC parallel to this one.
+
 ## Motivation
 
-Ultimately, at the end of the RFC, we want to be at a place where we can use decorators for clean and declarative classes that just use native javascript and using same import as the non-decorator equivalents. This is one of the last steps in making ember _just_ javascript.
+The Ember object-model has been on its way out for some time now, as native Javascript syntax has begun filling in the cracks and providing better, more standardized ways for us to define classes. The Native Class RFC from earlier this year was the first step toward deprecating and ultimately removing the object-model, and while we likely won't be able to fully remove it any time soon, one goal of Ember Octane will be to enable developers to write Ember apps using "just Javascript".
+
+The Octane programming model itself will help with this goal. Tracked properties will make usage of classic features, such as observers/computeds, much less necessary, and Glimmer components will _not_ extend from classic classes - native syntax will be the only option for defining them.
+
+However, there are aspects of the classic programming model which will be necessary for some time, including:
+
+* **Computed properties** and the change notification system that they rely on are incredibly common in Ember apps, and it will be a very long time before they can be fully converted to tracked properties throughout the ecosystem. Addons like Ember Data will need to be fully rewritten progressively, and being able to convert to native class syntax in the meantime will be incredibly useful.
+* **Actions** will still be part of Glimmer components, routes, and controllers for some time, and users will need a declarative way to define them.
```

Comment:
why is this? won't actions 'just' be functions?
or do you mean specifically the string access? `{[action 'actionName'}}` -- if so, wouldn't it be better to codemod to `{{action this.actionname}}`?

Reply:
or do you just mean that the string access of actions only needs to exist in the classic model?

Reply:
Personally, I think this is a better pattern. However, it won't work for Controllers or Routes, those would need to be rethought regardless because they do use `sendAction` internally. We should mention this in alternatives though, and tbh I think we should encourage this usage in templates so we can make the eventual switch.